File: tests/CodeAnalysis.CSharp.Tests/CodeAnalysis/CSharp/SyntaxFactsExtensionsTests.cs

using Microsoft.CodeAnalysis.CSharp;

namespace Altemiq.CodeAnalysis.CSharp
{
    public class SyntaxFactsExtensionsTests
    {
        [Test]
        [Arguments("int", "@int")]
        [Arguments("var", "@var")]
        [Arguments("other", "other")]
        public async Task Escape(string text, string expected)
        {
            _ = await Assert.That(SyntaxFacts.EscapeKeyword(text)).IsEqualTo(expected);
        }
    }
}

File: tests/CodeAnalysis.VisualBasic.Tests/CodeAnalysis/VisualBasic/SyntaxFactsExtensionsTests.vb
Imports Microsoft.CodeAnalysis.VisualBasic

Namespace CodeAnalysis.VisualBasic

    Public Class SyntaxFactsExtensionsTests
        <Test>
        <Arguments("Integer", "[Integer]")>
        <Arguments("Type", "[Type]")>
        <Arguments("Other", "Other")>
        Public Shared Async Function Escape(text As String, expected As String) As Task
            Dim unused = Await Assert.That(EscapeKeyword(text)).IsEqualTo(expected)
        End Function
    End Class
End Namespace